src/main/java/io/cdap/plugin/SCPRemotetoRemoteAction.java

```diff
@@ -0,0 +1,293 @@
+    }
+
+    @Override
+    public void run(ActionContext context) throws JSchException, IOException {
+        LOG.debug(String.format("Running the 'run' method of the %s plugin.", PLUGIN_NAME));
+        config.validate();
+
+        byte[] key = config.getPrivateKey();
+        byte[] passphrase = config.getPassphrase();
+        JSch jsch = new JSch();
+        jsch.addIdentity("key", key, null,passphrase);
+        Session session = jsch.getSession(config.getUserNameBastion(),config.getHostBastion(),config.getPort());
+
+        session.setConfig("StrictHostKeyChecking", "no");
+        session.connect();
+        ChannelExec channel = (ChannelExec) session.openChannel("exec");
+
+        String userA = config.getUserNameA();
+        String hostA = config.getHostA();
+        String source = config.getSource();
+        String pathSource = userA+"@"+hostA+":"+source;
+
+        String userB = config.getUserNameB();
+        String hostB = config.getHostB();
+        String dest= config.getDest();
+        String pathDest = userB+"@"+hostB+":"+dest;
+
+        String dirFlag = config.getDirFlag();
+        String compressFlag = config.getCompressFlag();
+        String verboseFlag = config.getVerboseFlag();
+
+        //Host A -> Host B
+        channel.setCommand("scp "+compressFlag+ " "+verboseFlag+
+            " " +dirFlag +" " +pathSource +" " +pathDest);
+
+        StringBuilder outputBuffer = new StringBuilder();
+        StringBuilder errorBuffer = new StringBuilder();
+
+        InputStream in = channel.getInputStream();
+        InputStream err = channel.getExtInputStream();
+
+        channel.connect();
+
+        byte[] tmp = new byte[1024];
+        while (true) {
+            while (in.available() > 0) {
+                int i = in.read(tmp, 0, 1024);
+                if (i < 0) break;
+                outputBuffer.append(new String(tmp, 0, i));
+            }
+            while (err.available() > 0) {
+                int i = err.read(tmp, 0, 1024);
+                if (i < 0) break;
+                errorBuffer.append(new String(tmp, 0, i));
+            }
+            if (channel.isClosed()) {
+                if ((in.available() > 0) || (err.available() > 0)) continue;
+                System.out.println("exit-status: " + channel.getExitStatus());
+                break;
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (Exception ignored) {
+            }
+        }
+
+        LOG.debug("output: " + outputBuffer.toString());
+        LOG.info("info: " + errorBuffer.toString());
+
+        channel.disconnect();
+        session.disconnect();
+
+    }
+
+    /**
+     * The config class for {@link SCPRemotetoRemoteAction} that contains all properties that need to be filled in by
+     * the user when building a Pipeline.
+     */
+    public static class SCPRemotetoRemoteActionConfig extends PluginConfig {
+
+        @Description("Hostname or IP Address of the SSH server.")
```

Comment:
Wrong alignment. CDAP codebase uses 2 spaces instead of 4 for indentation.

Reply:
I believe this is the alignment on the Example Action I used as a base. I can change it though